wordpress-mobile/WordPress-Android
# [Dependency Updates] Update `googlePlayServicesAuthVersion` to 20.4.1

Parent #17567

This PR update `googlePlayServicesAuthVersion` to [20.4.1](https://developers.google.com/android/guides/releases#january_19_2023).

-----

PS: @zwarm I added you as the main reviewer, not so randomly, since I just wanted someone from the WordPress team to be aware of and sign-off on that change for WPAndroid. I also added the @wordpress-mobile/apps-infrastructure team, but this in done only for monitoring purposes, as such, I am not expecting any active review from that team. Thus, feel free to merge this PR if you deem so.

-----

To test:

1. See the dependency tree diff result and verify correctness.
2. Thoroughly smoke test any auth related functionality on both, the WordPress and Jetpack apps, and see if they both work as expected.
3. In addition to the above smoke test, you can expand the below and follow the inner and more explicit test steps within (although verifying on `WordPress Google Sign-in` should be enough):

<details>
  <summary>WordPress Google Sign-in</summary> 

**Prerequisite**: If you don't have access to upload an`.aab` to Google Play Store for `WordPress - Website Builder`, post a request to `+systemsrequests`.

- Check-out this branch and edit the `version.properties` file to pick a dummy `versionName` and `versionCode`, preferable using a higher `versionCode` to the existing one.
    - I recommend using a `versionCode` of `1000000001` and above.
    - I recommend using an explicit `versionName` like `deps-update-google-play-services-20-4-1` (similar to the name of this branch itself)
- Build the release `.aab` locally using this command: `./gradlew bundleWordPressVanillaRelease`
- Upload the locally generated `.aab`, found within the `WordPress/build/outputs/bundle/wordpressVanillaRelease/` build folder, and into Google Play Console (`WordPress - Website Builder`) via: `App Bundle Explorer` > `Upload new version` (top-right)
     - Wait for the upload to complete, then via this newly uploaded app version, navigation to the `Downloads`, on the `Assets` section, find the `Signed, universal APK` and click the `Download` icon (middle-right) to download the signed `.apk` locally.
     - After having this signed `.apk` downloaded locally, go ahead and click on the `Delete app bundle` to delete the this app bundle from the list of available app bundles.
- Install this download signed `.apk` using this command: `adb install -r Downloads/wpandroid-deps-update-google-play-services-20-4-1-Signed.apk` (note that you can use `./tools/rename_apk_aab.sh` to rename the download signed `.apk`)
- If you were already logged-in to WordPress, log-out and log-in again using the `Continue with Google` button (note that you need to use a non `A8C` account to be able to complete this flow).
- Verify that the Google sign-in works, that you have successfully logged-in and are able to use the app as expected.

</details>

<details>
  <summary>Jetpack Google Sign-in</summary> 

**Prerequisite**: If you don't have access to upload an`.aab` to Google Play Store for `Jetpack - Website Builder`, post a request to `+systemsrequests`.

- Check-out this branch and edit the `version.properties` file to pick a dummy `versionName` and `versionCode`, preferable using a higher `versionCode` to the existing one.
    - I recommend using a `versionCode` of `1000000001` and above.
    - I recommend using an explicit `versionName` like `deps-update-google-play-services-20-4-1` (similar to the name of this branch itself)
- Build the release `.aab` locally using this command: `./gradlew bundleJetpackVanillaRelease`
- Upload the locally generated `.aab`, found within the `WordPress/build/outputs/bundle/jetpackVanillaRelease/` build folder, and into Google Play Console (`Jetpack - Website Builder`) via: `App Bundle Explorer` > `Upload new version` (top-right)
     - Wait for the upload to complete, then via this newly uploaded app version, navigation to the `Downloads`, on the `Assets` section, find the `Signed, universal APK` and click the `Download` icon (middle-right) to download the signed `.apk` locally.
     - After having this signed `.apk` downloaded locally, go ahead and click on the `Delete app bundle` to delete the this app bundle from the list of available app bundles.
- Install this download signed `.apk` using this command: `adb install -r Downloads/jpandroid-deps-update-google-play-services-20-4-1-Signed.apk` (note that you can use `./tools/rename_apk_aab.sh` to rename the download signed `.apk`)
- If you were already logged-in to Jetpack, log-out and log-in again using the `Continue with Google` button (note that you need to use a non `A8C` account to be able to complete this flow).
- Verify that the Google sign-in works, that you have successfully logged-in and are able to use the app as expected.

</details>

-----

## Regression Notes

1. Potential unintended areas of impact

    - Potential breakage or misbehaviour on any or all auth related app functionalities, like Google sign-in.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

PR submission checklist:
